### PR TITLE
Support #531 skip launching app activity if autoLaunch is set to false

### PR DESCRIPTION
**Pull request by rajdeepv**
Support #531

### CODE REVIEW

**Comment by rajdeepv** [comment]
Support #531

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt [hunk 1]
```diff
@@ -34,9 +34,12 @@ class CreateSession : RequestHandler<SessionParams, Session> {
         try {
             val parsedCaps = parseCapabilities(params.capabilities.firstMatch,
                     params.capabilities.alwaysMatch)
-            val activityName = parsedCaps["appActivity"] as? String
-                    ?: throw InvalidArgumentException("appActivity capability is mandatory")
-            startActivity(parsedCaps["appPackage"] as? String, activityName)
+            val shouldLaunchApp = (parsedCaps["autoLaunch"] ?: true) as Boolean
+            if (shouldLaunchApp) {
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt]: Was this change tested? As far as I remember the driver verifies that the given activity is running after the session has been started

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt [hunk 2]
```diff
@@ -34,9 +34,12 @@ class CreateSession : RequestHandler<SessionParams, Session> {
...
+                val activityName = parsedCaps["appActivity"] as? String
```
Comment (mykola-mokhnach) [inline espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt]: is it safe to ignore this check. Are there any other places in the code that assume this capability must exist?

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt (hunk 1 above)
Comment (rajdeepv) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt]: Yes, I give pass it the home package (launcher package) and '*' activity and it seems happy

Comment (mykola-mokhnach) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt]: Perhaps it makes sense to skip activity startup verification if  `autoLaunch ` is set. The "real" package cap value is needed for other API calls

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt (hunk 2 above)
Comment (rajdeepv) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt]: No, there is no other place that checks 'appActivity'

File: espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt (hunk 1 above)
Comment (rajdeepv) [reply espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/CreateSession.kt]: Yes, added

File: lib/driver.js [hunk 3]
```diff
@@ -300,8 +300,9 @@ class EspressoDriver extends BaseDriver {
 
     // launch espresso and wait till its online and we have a session
     await this.espresso.startSession(this.caps);
-    await this.adb.waitForActivity(this.caps.appWaitPackage, this.caps.appWaitActivity, this.opts.appWaitDuration);
-
+    if (this.caps.autoLaunch) {
```
Comment (mykola-mokhnach) [inline lib/driver.js]: `if (this.caps.autoLaunch === false)`

Comment (rajdeepv) [reply lib/driver.js]: done

**Comment by mykola-mokhnach** [comment]
Thanks, looks good to me

**Comment by rajdeepv** [comment]
Thank you @mykola-mokhnach  for your time :)

**Pull request merged**